flexmeasures_entsoe/utils.py

```diff
@@ -157,27 +157,57 @@ def abort_if_data_empty(data: Union[pd.DataFrame, pd.Series]):
         raise click.Abort
 
 
-def parse_from_and_to_dates_default_today_and_tomorrow(
-    from_date: Optional[datetime], to_date: Optional[datetime], country_timezone: str
+def abort_if_data_incomplete(
+    data: Union[pd.DataFrame, pd.Series],
+    from_time: pd.Timestamp,
+    until_time: pd.Timestamp,
+    resolution: pd.Timedelta,
+):
+    expected_periods = int((until_time - from_time) / resolution)
+    if len(data) < expected_periods:
+        click.secho(
+            f"Result is incomplete. Expected {expected_periods} periods but got {len(data)}. Probably ENTSO-E does not provide these forecasts yet ...",
+            **MsgStyle.ERROR,
+        )
+        raise click.Abort
+
+
+def parse_from_and_to_dates(
+    from_time: Optional[datetime],
+    until_time: Optional[datetime],
+    country_timezone: str,
+    default_to="today-and-tomorrow",  # Can be "tomorrow" or "today"
 ) -> Tuple[datetime, datetime]:
     """
     Parse CLI options (or set default to today and tomorrow)
```

Comment:
```suggestion
    Parse CLI options for start and end date (or set default to today and tomorrow) for inout to entsoe-py
    Note: we expect only dates as input here, and until_date is inclusice, so we extend it with 24h - so if from_date is equal to until_date, we return 00:00 and 24:00 of that day.
```

I believe much confusion in this PR came from this fact, which was under-documented. Also, I made you rename `from_date` to `from_time` (same with `until_*`). But you could also have checked what `parse_from_and_to_dates()` is doing.


Please rename `from_time` and `until_time` back again.

Reply:
Noted